PWGCF/FemtoUniverse/Tasks/femtoUniversePairTaskTrackTrackMultKtExtended.cxx

```diff
@@ -258,44 +258,78 @@ struct femtoUniversePairTaskTrackTrackMultKtExtended {
     // ConfNsigmaCombined -> TPC and TOF Sigma (combined) for momentum > ConfTOFpMin
 
     if (mom < twotracksconfigs.ConfTOFpMin) {
-      return TMath::Abs(nsigmaTPC) < twotracksconfigs.ConfNsigmaTPC;
+      if (TMath::Abs(nsigmaTPC) < twotracksconfigs.ConfNsigmaTPC) {
+        return true;
+      } else {
+        return false;
+      }
     } else {
-      return TMath::Hypot(nsigmaTOF, nsigmaTPC) < twotracksconfigs.ConfNsigmaCombined;
+      if (TMath::Hypot(nsigmaTOF, nsigmaTPC) < twotracksconfigs.ConfNsigmaCombined) {
+        return true;
+      } else {
+        return false;
+      }
     }
+
+    return false;
   }
 
   /// TPC Kaon Sigma selection (stricter cuts for K+ and K-) -- based on Run2 results
   bool IsKaonNSigma(float mom, float nsigmaTPCK, float nsigmaTOFK)
   {
     if (twotracksconfigs.IsKaonNsigma == true) {
       if (mom < 0.4) {
-        return TMath::Abs(nsigmaTPCK) < 2;
+        if (TMath::Abs(nsigmaTPCK) < 2) {
+          return true;
+        } else {
+          return false;
+        }
       } else if (mom > 0.4 && mom < 0.45) {
-        return TMath::Abs(nsigmaTPCK) < 1;
+        if (TMath::Abs(nsigmaTPCK) < 1) {
+          return true;
+        } else {
+          return false;
+        }
       } else if (mom > 0.45 && mom < 0.8) {
-        return (TMath::Abs(nsigmaTPCK) < 3 && TMath::Abs(nsigmaTOFK) < 2);
+        if (TMath::Abs(nsigmaTPCK) < 3 && TMath::Abs(nsigmaTOFK) < 2) {
+          return true;
+        } else {
+          return false;
+        }
       } else if (mom > 0.8 && mom < 1.5) {
-        return (TMath::Abs(nsigmaTPCK) < 3 && TMath::Abs(nsigmaTOFK) < 1.5);
+        if (TMath::Abs(nsigmaTPCK) < 3 && TMath::Abs(nsigmaTOFK) < 1.5) {
+          return true;
+        } else {
+          return false;
+        }
       } else {
         return false;
       }
     } else {
       return IsNSigma(mom, nsigmaTPCK, nsigmaTOFK);
     }
+
+    return false;
   }
 
   /// TPC Deuteron Sigma selection
   bool IsDeuteronNSigma(float mom, float nsigmaTPCDe, float nsigmaTOFDe)
   {
-    if (mom > deuteronconfigs.ConfPLowDe && mom < deuteronconfigs.ConfPHighDe) {
-      if (mom < deuteronconfigs.ConfTOFpMinDe) {
-        return TMath::Abs(nsigmaTPCDe) < deuteronconfigs.ConfNsigmaTPCDe;
+    if (mom < deuteronconfigs.ConfTOFpMinDe) {
```

Comment:
OK. Here you indeed change the logics a bit. You no longer check for pLow and pHigh before checking TOFpMin and nsigmas. Probably this is the one true fix that enables you to get more particles.

Could you also here use the convention `return TMath::Abs(nsigmaTPCDe) < deuteronconfigs.ConfNsigmaTPCDe` instead of repeating if...else ? Same for line 325. Trust me, it is exactly the same logics and it is not the cause of your problems :)